index.js

```diff
@@ -14,4 +14,91 @@ module.exports = (robot, _, Settings = require('./lib/settings')) => {
       return Settings.sync(context.github, context.repo())
     }
   }
+
+  robot.on('installation_repositories.added', installSync)
+
+  async function installSync (context) {
+    const payload = context.payload
+    // getting all the repos that are adding
+    const repoAddedId = await payload.repositories_added
+    const repoAddedIds = []
+    repoAddedId.forEach(async function (value) {
+      await repoAddedIds.push(value.id)
+    })
+    repoAddedIds.forEach(async function (value) {
+      const result = await context.github.repos.getById({id: value})
+      const owner = result.data.owner.login
+      const repoName = result.data.name
+      // As context.repo() was undefined so had to convert it into object
+      const repo = {
+        owner: owner,
+        repo: repoName
+      }
+      // repo should have a .github folder
+      const path = '.github'
+      try {
+        const repoInfo = await context.github.repos.getContent({owner: owner, repo: repoName, path: path})
+
+        const FILE_NAME = repoInfo.data.find(async file => {
+          if (file.name === 'settings.yml') {
+            return file.name
+          } else {
+            try {
+              const reference = await context.github.gitdata.getReference({owner: owner, repo: repoName, ref: 'heads/master'})
+              const refData = reference.data
+              const sha = refData.object.sha
+              try {
+                const getRepo = await context.github.repos.get({owner: owner, repo: repoName})
+                await context.github.gitdata.createReference({owner: owner, repo: repoName, ref: 'refs/heads/probot', sha: sha})
+                // setting the template of file
+                const string = 'repository: \n' +
```

Comment:
Can you please move this into a separate file? It'd also be good to use a [template literal]() to avoid all of the `\n` characters. In the separate file, you can do:

```js
module.exports = getRepo => `
  name: ${getRepo.data.name}
`
```

Reply:
Ooooo... I am not too proud to admit that I just learned a thing. Thanks! 🎉